zip-0311.rst

```diff
@@ -2,7 +2,262 @@
+and does not reveal:
+
+- Carol's payment address.
+- Any of Carol's other diversified payment addresses.
+- Any of Carol's other transactions.
+
+Donation drive
+--------------
+
+Diana is a well-known individual with a public Sapling payment address (for receiving
+donations). She runs a matching donation drive to the non-profit Totally Legit Bunnies,
+and wants to prove to her followers that she correctly matched their dontations. Diana
+can create a payment disclosure, and post it to her social media account, that reveals:
+
+- The correct amount was sent to the Totally Legit Bunnies' recipient address in the
+  given transaction.
+- The sender of the transaction is the payment address that is associated with Diana.
+
+and does not reveal:
+
+- Any of Diana's other diversified payment addresses.
+- Any of Diana's other transactions.
+
+Merchant dispute
+----------------
+
+Edward goes to CoinCafe to buy a sandwich, and sends funds from his Sapling address to
+CoinCafe's transparent address. They can see a payment has been made, but claim that they
+have not received a payment from Edward. He can create a payment disclosure that reveals:
+
+- Edward was the sender of the given transaction.
+
+and does not reveal:
+
+- Edward's payment address.
+- Any of Edward's other diversified payment addresses.
+- Any of Edward's other transactions.
+
+Shielded withdrawals from an exchange
+-------------------------------------
+
+CorpBux is an exchange with a transparent hot wallet, and enables customers to withdraw
+their funds to Sapling addresses. They can create a payment disclosure, that they can give
+to their customers, that reveals:
+
+- The correct amount was sent to the customer's recipient address in the given transaction.
+- CorpBux was the sender of the given transaction.
+
+
+Requirements
+============
+
+- The payment disclosure may disclose the contents of zero or more Sapling outputs within
+  a single transaction. (Zero outputs is useful for t2z transactions.)
+- Payment disclosures are non-malleable, and can only be created by a sender of the
+  transaction.
+- A payment disclosure may be tied to a challenge, creating an interactive proof.
+- Senders with a Sapling address may choose to provably reveal that address as part of the
+  payment disclosure.
+- Senders are not required to remember any ephemeral data from the creation of a transaction
+  in order to produce a payment disclosure for that transaction. (Performance may be
+  improved if they do cache witness data for spent notes.)
+
+
+Specification
+=============
+
+Payment disclosure data structure
+---------------------------------
+
+A payment disclosure has the following fields:
+
+- `txid`: Transaction id for the given transaction `tx`.
+
+- `msg`: A message field, which could contain a challenge value from the party to whom
+  the payment disclosure is directed.
+
+- `sapling_outputs`: A vector of the Sapling Outputs that we are disclosing
+  [0..tx.shieldedOutputs.len()]
```

Comment:
This is inclusive of 0 and exclusive of `tx.shieldedOutputs.len()`, but the notation hasn't been defined.